tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,770 @@
+        }
+
+        void reset() {
+            this.inputRealmLocation = null;
+            this.outputRealmLocation = null;
+            this.FileSystemRealmName = null;
+            this.credentialStore = null;
+            this.secretKeyAlias = null;
+            this.hashCharset = null;
+            this.hashEncoding = null;
+            this.levels = 0;
+        }
+    }
+
+    @Override
+    public void execute(String[] args) throws Exception {
+        setStatus(GENERAL_CONFIGURATION_ERROR);
+        cmdLine = parser.parse(options, args, false);
+        setEnableDebug(cmdLine.hasOption(DEBUG_PARAM));
+        if (cmdLine.hasOption(HELP_PARAM)) {
+            help();
+            setStatus(ElytronTool.ElytronToolExitStatus_OK);
+            return;
+        }
+        if (cmdLine.hasOption(SILENT_PARAM)) {
+            silentMode = true;
+        }
+        if (cmdLine.hasOption(SUMMARY_PARAM)) {
+            summaryMode = true;
+            summaryString = new StringBuilder();
+            summaryString.append(String.join("", Collections.nCopies(SUMMARY_WIDTH, "-")));
+            summaryString.append(System.getProperty("line.separator"));
+            summaryString.append("Summary for execution of Elytron-Tool command FileSystemEncrypt");
+            summaryString.append(System.getProperty("line.separator"));
+            summaryString.append(String.join("", Collections.nCopies(SUMMARY_WIDTH, "-")));
+            summaryString.append(System.getProperty("line.separator"));
+        }
+        printDuplicatesWarning(cmdLine);
+
+        String realmNameOption = cmdLine.getOptionValue("r");
+        String inputRealmLocationOption = cmdLine.getOptionValue("i");
+        String outputRealmLocationOption = cmdLine.getOptionValue("o");
+        String credentialStoreOption = cmdLine.getOptionValue("c");
+        String secretKeyAliasOption = cmdLine.getOptionValue("k");
+        String hashCharsetOption = cmdLine.getOptionValue("h");
+        String hashEncodingOption = cmdLine.getOptionValue("e");
+        String levelsOption = cmdLine.getOptionValue("l");
+        String bulkConvert = cmdLine.getOptionValue("b");
+
+        if (bulkConvert == null) {
+            if (realmNameOption == null) {
+                realmNameOption = DEFAULT_FILESYSTEM_REALM_NAME;
+            }
+
+            Descriptor descriptor = new Descriptor();
+            descriptor.setFileSystemRealmName(realmNameOption);
+            descriptor.setOutputRealmLocation(Paths.get(outputRealmLocationOption).toString());
+            descriptor.setInputRealmLocation(Paths.get(inputRealmLocationOption).toString());
+            if (hashCharsetOption == null) {
+                descriptor.setHashCharset(StandardCharsets.UTF_8);
+            } else {
+                try {
+                    descriptor.setHashCharset(Charset.forName(hashCharsetOption));
+                } catch (UnsupportedCharsetException e) {
+//                    Error
```

Comment:
Looks like error handling needs to be added here and below (there should be some examples in the existing commands).

Reply:
Added